core/src/main/java/io/aiven/klaw/service/BaseOverviewService.java

```diff
@@ -224,9 +223,19 @@ protected List<AclOverviewInfo> applyFiltersAclsForSOT(
 
     List<AclOverviewInfo> aclList = new ArrayList<>();
     AclOverviewInfo mp;
+    List<AclRequests> disableShowDeleteAcl;
 
     for (Acl aclSotItem : aclsFromSOT) {
+      disableShowDeleteAcl =
+          aclRequestsRepo.findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname(
+              tenantId,
+              aclSotItem.getEnvironment(),
+              RequestStatus.DELETED.value,
+              aclSotItem.getTopicname());
       if (aclSotItem.getAclip() != null || aclSotItem.getAclssl() != null) {
+        boolean deleteRequestExists =
+            disableShowDeleteAcl.stream()
+                .anyMatch(item -> item.getReq_no().equals(aclSotItem.getReq_no()));
```

Comment:
In terms of matching I see the problem, 
The req_no is unique for the ACL req but the ACL references the other acl by using the otherParams field. So it is a confusing and not ideal situtation.

however in terms of resolving this for now, you should be able to match against otherParams.

Reply:
> In terms of matching I see the problem, The req_no is unique for the ACL req but the ACL references the other acl by using the otherParams field. So it is a confusing and not ideal situtation.
> 
> however in terms of resolving this for now, you should be able to match against otherParams.

what do you think would be the best solution to this. 
And if I'm not wrong `otherParams` stores `AclIDs` to refer to other `AclRequests` right?

Reply:
Yes otherParams stores the request number to refer to another ACL you just need to add the tenantId to make that a full ACLID.

The best solution long term I think would be to add a new column, but I think you can just use the returned list of Ids and put them into a Set<Integer> and check if the entry exists, setDeleteRequestExists(set.contains(requestNum);